app/response/utils.py

```diff
@@ -326,104 +354,228 @@ def process_email_template_request(request_id, data):
     page = flask_request.host_url.strip('/') + url_for('request.view', request_id=request_id)
     agency_name = Requests.query.filter_by(id=request_id).first().agency.name
     email_template = os.path.join(current_app.config['EMAIL_TEMPLATE_DIR'], data['template_name'])
-    # process email template for extension
-    if data['type'] == 'extension_email':
-        # if data['extension'] exists, use email_content as template with specific extension email template
-        try:
-            extension = data['extension']
-            default_content = False
-            content = data['email_content']
-            # calculates new due date based on selected value if custom due date is not selected
-            new_due_date = _get_new_due_date(request_id,
-                                             extension['length'],
-                                             extension['custom_due_date']).strftime('%A, %b %d, %Y')
-            reason = extension['reason']
-        # use default_content in response template
-        except KeyError:
-            default_content = True
-            content = None
-            new_due_date = ''
-            reason = ''
-        return render_template(email_template,
-                               default_content=default_content,
-                               content=content,
-                               request_id=request_id,
-                               agency_name=agency_name,
-                               new_due_date=new_due_date,
-                               reason=reason,
-                               page=page)
-    # process email template for file upload
-    if data['type'] == 'file_upload_email':
-        # create a dictionary of filenames to be passed through jinja to email template
-        files_links = {}
-        # if data['files'] exists, use email_content as template with specific file email template
-        try:
-            files = data['files']
-            default_content = False
-            content = data['email_content']
-        # use default_content in response template
-        except KeyError:
-            files = []
+    # set a dictionary of email types to handler functions to handle the specific response type
+    handler_for_type = {
+        response_type.EXTENSION_EMAIL: _extension_email_handler,
```

Comment:
Just use `response_type.EXTENSION` and so on for now (and update the calls on the front end).